libs/core/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts

```diff
@@ -250,66 +284,73 @@ export class CalendarDayViewComponent<D> implements OnInit, OnChanges, Focusable
+        ) {
             this._buildDayViewGrid();
         }
         if (changes['startingDayOfWeek']) {
             this._refreshShortWeekDays();
         }
     }
+
     /**
-     * Function for selecting a date on the calendar. Typically called when a date is clicked, but can also be called programmatically.
-     * @param day CalendarDay object to be selected.
-     * @param event
+     * @hidden
+     * Selects a date in single mode, toggles selection in multi mode, or sets the range in range mode.
+     * Handles the selection of a date based on the calendar type.
+     * @param day The calendar day to be selected.
+     * @param event Optional mouse event for handling event propagation.
      */
     selectDate(day: CalendarDay<D>, event?: MouseEvent): void {
         if (event) {
-            /**
-             * There are some problems with popup integration. After clicking inside day component, the popover closes.
-             */
             event.stopPropagation();
             event.preventDefault();
         }
 
-        if (!day.disabled && !day.blocked) {
-            if (this.calType === 'single') {
-                /** Remove selections from other day and put selection to chosen day */
-                this._calendarDayList.forEach((_day) => (_day.selected = false));
-                day.selected = true;
-                this._selectedDate = day.date;
-                this.selectedDateChange.emit(day.date);
-            } else {
-                if (this._selectCounter === 0 || this._selectCounter === 2) {
-                    this._selectedRangeDate = { start: day.date, end: null };
-                    this.selectedRangeDateChange.emit(this.selectedRangeDate);
-                } else if (this._selectCounter === 1) {
-                    // Check if date picked is higher than already chosen, otherwise just reverse them
-                    if (this._dateTimeAdapter.compareDate(this.selectedRangeDate.start!, day.date) < 0) {
-                        this._selectedRangeDate = {
-                            start: this.selectedRangeDate.start,
-                            end: day.date
-                        };
-                    } else {
-                        this._selectedRangeDate = {
-                            start: day.date,
-                            end: this.selectedRangeDate.start
-                        };
-                    }
-                    this.selectedRangeDateChange.emit(this.selectedRangeDate);
-                }
-                this._changeSelectedRangeDays(this._selectedRangeDate, this._calendarDayList);
-            }
+        if (day.disabled || day.blocked) {
+            return;
         }
 
-        if (this.calType === 'range' && this.rangeHoverEffect && this._selectCounter === 1 && event) {
-            this._isOnRangePick = !this._isOnRangePick;
-        } else {
-            this._isOnRangePick = false;
+        switch (this.calType) {
+            case 'single':
```

Comment:
@dpavlenishvili  
Would it be better to create an enum for 'single', 'multi', and 'range' and use it inside the case statements? This would allow us to use the enum values instead of hardcoded strings, improving code maintainability and reducing the risk of errors due to typos.